### ISSUE TITLE
False positive errors with divmod timedelta

### ISSUE THREAD

**Issue by oefe** [microsoft/pyright#1629]
**Describe the bug**

Pylance reports "Type is incompatible" errors when using `divmod` with two `datetime.timedelta` parameters

**To Reproduce**

```python
from datetime import timedelta
hours, rest = divmod(timedelta(minutes=90), timedelta(hours=1))
```

Pylance reports the following errors:

```
Argument of type "timedelta" cannot be assigned to parameter "__x" of type "_N2@divmod" in function "divmod"
  Type "timedelta" is incompatible with constrained type "_N2"
Argument of type "timedelta" cannot be assigned to parameter "__y" of type "_N2@divmod" in function "divmod"
  Type "timedelta" is incompatible with constrained type "_N2"
```

**Expected behavior**
No errors. This is valid code.

**VS Code extension or command-line**
Using VS Code: 1.54.2, PyLance v2021.3.1, Python 3.9.1

**Additional context**

`divmod` is supported for `timedelta` objects, see [documentation](https://docs.python.org/3/library/datetime.html?highlight=timedelta#datetime.timedelta):

Operation | Result
------------ | -------------
q, r = divmod(t1, t2) | Computes the quotient and the remainder: q = t1 // t2(3) and r = t1 % t2. q is an integer and r is a timedeltaobject.

mypy 0.812 accepts the example without errors.

**Comment by erictraut** [comment]
Pyright is doing the correct thing here based on the information provided in the `builtins.pyi` typeshed stub.

Here's how the `divmod` function is defined:

```python
_N2 = TypeVar("_N2", int, float)
def divmod(__x: _N2, __y: _N2) -> Tuple[_N2, _N2]: ...
```

Note that the `_N2` TypeVar is constrained to be either an `int` or `float`, but `timedelta` does not derive from either of these classes.

It's interesting to note that if I define a local version of `divmod` by copying this definition out of `builtins.pyi` into my local test file, mypy then complains. That tells me that mypy is shipping with old versions of the typeshed stubs, has some special-case handling that overrides the type information for divmod, or has a bug. Digging into the mypy source, it appears that someone has added a special-case hack. We try very hard to avoid adding such special-case knowledge in pyright because the approach isn't scalable. The correct fix for this is to properly define the method in the `builtins.pyi` typeshed file.

I've submitted the following PR to the typeshed github repo that addresses this problem: https://github.com/python/typeshed/pull/5114. This fix will address the issue with all Python type checkers and will even allow mypy to remove the special-case handling that it includes currently.

**Comment by erictraut** [comment]
My change was accepted into the typeshed repo, and I've updated the typeshed stubs that ship with pyright, so this will be addressed in the next release.

**Comment by erictraut** [comment]
This change is included in pyright 1.1.122, which I just published. It will also be in the next release of pylance.
